### DISCUSSION TITLE
ConnectionParams vs ConnectTimeout

### DISCUSSION THREAD

**Discussion by thekurtovic** [h2zero/NimBLE-Arduino#663] [Q&A] [answered]
Hi,
I was reading through a recent thread which suggested that connection parameters could be increased in order to achieve 100 meter distances before attempting a connection.
#661 

The NimBLE_Client example has the following setup before calling the connection function.
`pClient->setConnectionParams(12,12,0,51);
pClient->setConnectTimeout(5);`
What is the difference between the connection timeout compared to the connection parameter's timeout? 
The example appears to have a 510 ms timeout from setConnectionParams(), but then has a 5 second timeout from setConnectionTimeout(). Since the former is also known as the "supervision timeout", my current understanding is:

- setConnectionTimeout() defines how long to wait after calling pClient->connect() before bailing out
- setConnectionParams() timeout defines how long to wait _after a connection is established_ before disconnecting due to inactivity

Is this correct? If so, would you recommend also increasing setConnectionTimeout() in order to achieve longer connection distances?
Thank you

**Comment by h2zero** [comment, marked as answer]
Sorry for the late reply, your understanding of those functions is correct. I would personally set them both higher, `setConnectionTimeout(30)` and `setConnectionParams(12,12,0,250)`
